app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -86,20 +96,80 @@ def proof_with_threatmetrix_if_needed(
         end
       end
 
-      def proof_resolution(applicant_pii:, timer:)
+      def proof_residential_address_if_needed(
+        applicant_pii:,
+        timer:,
+        double_address_verification:
+      )
+        return residential_address_unnecessary_result unless double_address_verification
+
+        timer.time('residential address') do
+          resolution_proofer.proof(applicant_pii)
+        end
+      end
+
+      def residential_address_unnecessary_result
+        Proofing::AddressResult.new(
+          success: true, errors: {}, exception: nil, vendor_name: 'ResidentialAddressNotRequired',
+        )
+      end
+
+      def resolution_unnecessary_result
+        Proofing::AddressResult.new(
+          success: true, errors: {}, exception: nil, vendor_name: 'ResolutionUnnecessary',
+        )
+      end
+
+      # TK: confirm vendor name with Team Ada
+      def resolution_cannot_pass
+        Proofing::AddressResult.new(
+          success: false, errors: {}, exception: nil, vendor_name: 'ResolutionCannotPass',
+        )
+      end
+
+      def proof_id_address_with_lexis_nexis_if_needed(applicant_pii:, timer:,
+                                                      residential_instant_verify_proof:)
+        return residential_instant_verify_proof if applicant_pii[:same_address_as_id] == 'true'
+        return resolution_cannot_pass unless residential_instant_verify_proof.success?
+
         timer.time('resolution') do
           resolution_proofer.proof(applicant_pii)
         end
       end
 
-      def proof_state_id_if_needed(
+      def aamva_irrelevant?(double_address_verification:, same_address_as_id:,
+                            should_proof_state_id:, instant_verify_proof:,
+                            residential_instant_verify_proof:)
+        return true unless should_proof_state_id
+        if double_address_verification == false &&
+           !user_can_pass_after_state_id_check?(instant_verify_proof)
+          return true
+        end
+        if double_address_verification == true && same_address_as_id == 'false' &&
+           !residential_instant_verify_proof.success?
+          return true
+        end
+        if double_address_verification == true && same_address_as_id == 'true' &&
+           !user_can_pass_after_state_id_check?(instant_verify_proof)
+          return true
+        end
```

Comment:
I wonder if we'd want to return `false` at the end. It seems like this will only ever return `true` or `nil`.

Reply:
We could return the conditional itself, and this could be a bit simpler. E.g. like the following.

```ruby
  unless should_proof_state_id
    true
  elsif double_address_verification == false || same_address_as_id == 'true'
    !user_can_pass_after_state_id_check?(instant_verify_proof)
  else
    !residential_instant_verify_proof.success?
  end
```

Reply:
Looking into these suggestions! 🙏🏻

Reply:
I refactored the conditional logic and the name.  Thanks, all!